Samples/Win32WebSocket/main.cpp

```diff
@@ -218,80 +218,77 @@ int main()
     HRESULT hr = HCWebSocketCreate(&websocket, message_received, binary_message_received, websocket_closed, nullptr);
     HCWebSocketSetBinaryMessageFragmentEventFunction(websocket, binary_message_fragment_received);
 
-    for (int iConnectAttempt = 0; iConnectAttempt < 10; iConnectAttempt++)
+    g_numberMessagesReceieved = 0;
+
+    XAsyncBlock* asyncBlock = new XAsyncBlock{};
+    asyncBlock->queue = g_queue;
+    asyncBlock->callback = [](XAsyncBlock* asyncBlock)
+    {
+        WebSocketCompletionResult result = {};
+        HCGetWebSocketConnectResult(asyncBlock, &result);
+
+        printf_s("HCWebSocketConnect complete: %d, %d\n", result.errorCode, result.platformErrorCode);
+        SetEvent(g_eventHandle);
+        delete asyncBlock;
+    };
+
+    printf_s("Calling HCWebSocketConnect...\n");
+    hr = HCWebSocketConnectAsync(url.data(), "", websocket, asyncBlock);
+    WaitForSingleObject(g_eventHandle, INFINITE);
+
+    uint32_t numberOfMessagesToSend = 1;
+    for (uint32_t i = 1; i <= numberOfMessagesToSend; i++)
     {
-        g_numberMessagesReceieved = 0;
+        char webMsg[150000];
+        for (uint32_t j = 0; j < sizeof(webMsg); ++j)
+        {
+            webMsg[j] = 'X';
+        }
+        webMsg[sizeof(webMsg) - 1] = '\0';
+        snprintf(webMsg, sizeof(webMsg), "Message #%d should be echoed!", i);
 
-        XAsyncBlock* asyncBlock = new XAsyncBlock{};
+        asyncBlock = new XAsyncBlock{};
         asyncBlock->queue = g_queue;
         asyncBlock->callback = [](XAsyncBlock* asyncBlock)
         {
             WebSocketCompletionResult result = {};
-            HCGetWebSocketConnectResult(asyncBlock, &result);
+            HCGetWebSocketSendMessageResult(asyncBlock, &result);
```

Comment:
>HCGetWebSocketSendMessageResult [](http://example.com/codeflow?start=12&length=31)

check hr here and below Result().  

in this file, i would add asserts for all the hrs just for extra checking